cabot/cabotapp/models.py

```diff
@@ -934,6 +951,50 @@ def _run(self):
         return result
 
 
+class TCPStatusCheck(StatusCheck):
+
+    @property
+    def check_category(self):
+        return "TCP Check"
+
+    @property
+    def description(self):
+        return 'Monitoring connection of {0}:{1}'.format(self.address, self.port)
+
+    address = models.CharField(
+        max_length=1024,
+        help_text='IP address or hostname to monitor',)
+    port = models.PositiveIntegerField(
+        help_text='Port to listen on',)
+    timeout = PositiveIntegerMaxField(
+        default=8,
+        max_value=16,
+        help_text='Timeout on idle connection after this many seconds',)
+
+    update_url = 'update-tcp-check'
+
+    icon = 'glyphicon glyphicon-text-wide'
+
+    '''
+    We wish to provide a method to ensure a TCP endpoint is still up. To
+    achieve this, we use the python socket library to connect to the TCP
+    service listening on the specified address and port. In other words,
+    if this call succeeds (i.e. returns without raising an exception and/or
+    timeing out), we can conclude that the TCP endpoint is valid.
+    '''
```

Comment:
nit: docstring indented after the function declaration please https://www.python.org/dev/peps/pep-0257/